plugins/libplugin-pay.c

```diff
@@ -1236,10 +1255,11 @@ void payment_continue(struct payment *p)
 	abort();
 }
 
-void payment_fail(struct payment *p)
+void payment_fail(struct payment *p, const char *reason)
 {
 	p->end_time = time_now();
 	p->step = PAYMENT_STEP_FAILED;
+	p->failreason = tal_steal(p, reason);
```

Comment:
Stealing without warning the caller is an error magnet; in this case, reason could be a string literal and it should work.

~~I'd suggest tal_dup_arr(...) here and label `reason` TAKES.~~

Actually, looking at the user: make payment_fail take a fmt string instead?